File: Example/第三方框架/第三方框架/Classes/Main/Controller/MainViewController.m
//
//  MainViewController.m
//  第三方框架
//
//  Created by 金亮齐 on 2017/3/2.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "MainViewController.h"
#import "HomeViewController.h"
#import "ProfileViewController.h"
#import "MainNavigationController.h"

@interface MainViewController ()

@end

@implementation MainViewController


+(void)initialize{

    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setUpChildVC];
    // Do any additional setup after loading the view.
}

- (void)setUpChildVC{
    
    /** 添加子控制器 */
    [self setupChildVc:[[HomeViewController alloc] init] title:@"首页" image:@"bottom_tab1_unpre" selectedImage:@"bottom_tab1_pre"];
    
//    [self setupChildVc:[[FoundViewController alloc] init] title:@"发现" image:@"found" selectedImage:@"foundSeletced" ];
//    
//    [self setupChildVc:[[MoreViewController alloc] init] title:@"更多" image:@"more" selectedImage:@"moreSeletced"];
//    
    [self setupChildVc:[[ProfileViewController alloc] init] title:@"我" image:@"bottom_tab2_unpre" selectedImage:@"bottom_tab2_pre"];
    
}

/* 初始化子控制器 **/
- (void)setupChildVc:(UIViewController *)vc title:(NSString *)title image:(NSString *)image selectedImage:(NSString *)selectedImage
{
    
    // 设置文字和图片
    vc.tabBarItem.title = title;
    vc.tabBarItem.image = [UIImage imageNamed:image];
    vc.tabBarItem.selectedImage = [UIImage imageNamed:selectedImage];
    // 添加为子控制器
    [self addChildViewController:vc];
    //     取消选中时的背景色
    vc.tabBarItem.selectedImage = [vc.tabBarItem.selectedImage imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
    
    //改变UITabBarItem字体颜色
    [[UITabBarItem appearance] setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys:navigationBarColor,UITextAttributeTextColor, nil] forState:UIControlStateSelected];
    
    
    //包装一个导航控制器，添加导航控制器为tabbarController的子控制器
    MainNavigationController *nav = [[MainNavigationController alloc] initWithRootViewController:vc];
    [self addChildViewController:nav];
    
}

@end

File: 项目/Swift项目/斗鱼(Swift)/斗鱼(Swift)/Classes/Main/Controller/MainViewController.swift
//
//  MainViewController.swift
//  斗鱼(Swift)
//
//  Created by 金亮齐 on 2017/5/7.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

import UIKit

class MainViewController: UITabBarController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        //MARK: - 点击时候文字的颜色
        let tabBarAppear = UITabBarItem.appearance()
        tabBarAppear.setTitleTextAttributes([NSForegroundColorAttributeName:UIColor.orange], for: UIControlState.selected)

        setUpAllChildViewControllers()
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - 设置控制器
    fileprivate func setUpAllChildViewControllers () {
        let normalImages = ["btn_home_normal","btn_live_normal","btn_column_normal","btn_user_normal"]
        let selectedImages = ["btn_home_selected","btn_live_selected","btn_column_selected","btn_user_selected"]
        
        let newsVC = HomeViewController()
        setUpNavRootViewCOntrollers(vc: newsVC, title: "首页", imageName: normalImages[0], selImage: selectedImages[0])
        
        let circleVC = LiveViewController()
        setUpNavRootViewCOntrollers(vc: circleVC, title: "直播", imageName: normalImages[1], selImage: selectedImages[1])
        
        let garageVC = UIViewController()
        setUpNavRootViewCOntrollers(vc: garageVC, title: "关注", imageName: normalImages[2], selImage: selectedImages[2])
        
        let profileVC = ProfileViewController()
        setUpNavRootViewCOntrollers(vc: profileVC, title: "我的", imageName: normalImages[3], selImage: selectedImages[3])
    }
    
    
    fileprivate func setUpNavRootViewCOntrollers(vc: UIViewController, title:String, imageName: String, selImage: String) {
        vc.title = title
        vc.tabBarItem.image = UIImage(named: imageName)
        vc.tabBarItem.selectedImage = UIImage(named: selImage)?.withRenderingMode(UIImageRenderingMode.alwaysOriginal)
        self.addChildViewController(MainNavController(rootViewController: vc))
    }
}
